python/mxnet/gluon/contrib/estimator/estimator.py

```diff
@@ -64,13 +64,17 @@ def __init__(self, net,
                  metrics=None,
                  initializer=None,
                  trainer=None,
-                 context=None):
+                 context=None,
+                 evaluation_loss=None):
 
         self.net = net
         self.loss = self._check_loss(loss)
         self._train_metrics = _check_metrics(metrics)
         self._add_default_training_metrics()
         self._add_validation_metrics()
+        self.evaluation_loss = None
```

Comment:
How about changing this line to `self.evaluation_loss = self.loss`. Then you can remove the if else clause in the `def evaluate_batch`

Reply:
If users need customized evaluation loss in the base estimator, they cannot do that any more.

Reply:
Why not? The semantics is simpler: `self.evaluation_loss` always refers to the loss used during evaluation. The proposed change is to simply initialize `self.evaluation_loss` with `self.loss` if the user does not specify `evaluation_loss` explicitly.

Reply:
Yes, you are right.